### PR TITLE
Fixes part of #591; when 2 calls to HasIndex reference same column

### PR DESCRIPTION
**Pull request by phkelley**
I was looking at what's required to move my project from EF 6.2 to EF Core, and one big change is the move from `[Index]` attributes to calls to `.HasIndex`. So I said, let me change to `.HasIndex` first, since that syntax is available in EF 6.2. Then when I switch to EF Core, it'll be a snap - almost no code changes.

But I hit a bug that happens when you use `HasIndex` to establish two indexes that both contain the same column. This bug was previously reported in #591.

Basically, since the data is stored internally in EF by column, a second index definition displaces the first, for those columns that are in both indexes. I have updated the logic to do a merge instead, and added a test case.

### CODE REVIEW

**Comment by Kantuz001** [comment]
i hope this will be merged

**Comment by ajcvickers** [comment]
@phkelley Thanks for the PR and sorry for the delay. This is now merged.
